test/suite/test_compact03.py

```diff
@@ -0,0 +1,155 @@
+    #    file.
+    # 4. Perform checkpoint to ensure overflow values are written on disk.
+    # 5. Delete middle ~90% of the normal values in the table.
+    # 6. Perform checkpoint so compact can find something to work with.
+    # 7. Call compact.
+    # 8. Get stats on compacted table expecting that there will no change in size given we have
+    #    overflow keys at the end of file.
+    # 9. Insert some normal values again. They will be written in the free extents in the middle
+    #    of the file. Therefore, expect no increase in file size.
+    #
+    # We want to have around 20000 leaf pages. With minimum 1KB page allocation size, the table
+    # is expected to have at least 25 MByte worth of data. We can then experiment with deleting
+    # range of keys in middle to test how comapction works.
+
+    normalValue = "abcde" * 10
+    overflowValue = "abcde" * 1000
+    nrecords = 400000 # To create ~25 MB table
+    expectedTableSize = 20 # Mbytes
+    nOverflowRecords = 5000
+
+    # Return the size of the file
+    def getSize(self):
+        # To allow this to work on systems without ftruncate,
+        # get the portion of the file allocated, via 'statistics=(all)',
+        # not the physical file size, via 'statistics=(size)'.
+        cstat = self.session.open_cursor(
+            'statistics:' + self.uri, None, 'statistics=(all)')
+        sz = cstat[stat.dsrc.block_size][2]
+        cstat.close()
+        return sz
+
+    # Create a table, add keys with both big and small values.
+    def test_compact03(self):
+
+        mb = 1024 * 1024
+        # 1. Create a table with relatively small page size.
+        params = 'key_format=i,value_format=S,' + self.fileConfig
+        self.session.create(self.uri, params)
+        c = self.session.open_cursor(self.uri, None)
+        for i in range(self.nrecords):
+            c[i] = self.normalValue
+        c.close()
+
+        # 2. Checkpoint and get stats on the table to confirm the size.
+        self.session.checkpoint()
+        sizeWithoutOverflow = self.getSize()
+        self.pr('After populate ' + str(sizeWithoutOverflow // mb) + 'MB')
+        self.assertGreater(sizeWithoutOverflow, self.expectedTableSize * mb)
+
+        # 3. Add overflow values.
+        c = self.session.open_cursor(self.uri, None)
+        for i in range(self.nOverflowRecords):
+            c[i + self.nrecords] = self.overflowValue
+        c.close()
+
+        # 4. Perform checkpoint to ensure overflow values are written to the disk.
+        self.session.checkpoint()
+        sizeWithOverflow = self.getSize()
+        self.pr('After inserting overflow values ' + str(sizeWithoutOverflow // mb) + 'MB')
+        self.assertGreater(sizeWithOverflow, sizeWithoutOverflow)
+
+        # 5. Delete middle ~90% of the normal values in the table.
+        c = self.session.open_cursor(self.uri, None)
+        for i in range((self.nrecords // 100) * 5, (self.nrecords // 100) * 95):
```

Comment:
Can you do a fast truncate here with the start and end key values?

Reply:
I have added a scenario where we use truncate to delete keys. See 0ac13fe.